src/Microsoft.Graph.Core/Tasks/LargeFileUploadTask.cs

```diff
@@ -72,14 +74,10 @@ private IUploadSession ExtractSessionFromParsable(IParsable uploadSession)
             if (!uploadSession.GetFieldDeserializers().ContainsKey("uploadUrl"))
                 throw new ArgumentException("The Parsable does not contain the 'uploadUrl' property");
 
-            var uploadSessionType = uploadSession.GetType();
-
-            return new UploadSession()
-            {
-                ExpirationDateTime = uploadSessionType.GetProperty("ExpirationDateTime").GetValue(uploadSession, null) as DateTimeOffset?,
-                NextExpectedRanges = uploadSessionType.GetProperty("NextExpectedRanges").GetValue(uploadSession, null) as List<string>,
-                UploadUrl = uploadSessionType.GetProperty("UploadUrl").GetValue(uploadSession, null) as string
-            };
+            // convert to local type as we don't have the type info for the upload session just that it implements IParsable
+            var uploadSessionStream = KiotaSerializer.SerializeAsStream(CoreConstants.MimeTypeNames.Application.Json, uploadSession, false);// just in case there's a backing store 
```

Comment:
Missing using

Reply:
not sure what you mean here. Any chance you can confirm?

Reply:
stream should be disposed